syne_tune/optimizer/schedulers/searchers/multi_fidelity_kde_searcher.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from typing import Dict, Optional, List
+import logging
+import numpy as np
+
+from syne_tune.optimizer.schedulers.searchers.kde_searcher import KernelDensityEstimator
+
+__all__ = ['MultiFidelityKernelDensityEstimator']
+
+logger = logging.getLogger(__name__)
+
+
+class MultiFidelityKernelDensityEstimator(KernelDensityEstimator):
```

Comment:
Are you sure this also works with the async schedulers?

If so, shall we use this as another baseline in the JMLR paper, i.e. ASHA with model-based decisions by KDE?

Reply:
Yes we could use it with asynchronous schedulers, the same way hyperopt does it, by ignoring pending evaluations.